airflow/cli/cli_config.py

```diff
@@ -382,11 +382,19 @@ def string_lower_type(val):
     ),
     action="store_true",
 )
+# TODO: Deprecate in backfill command in favor of --treat-dag-id-as-regex
```

Comment:
Mention it in `help` as other deprecated parameters do

Reply:
Something like `will be later deprecated in favor of --treat-dag-id-as-regex`? (I'd like to avoid touching `backfill` logic as part of this PR)